src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndex(index);
+        if (size == 0) {
+            return null;
+        }
+        if (index == 0) {
+            return head.item;
+        }
+        if (index > 0) {
+            Node<T> currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+            return currentNode.item;
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = get(index);
+        if (size > 0 && index < (size - 1) && index > 0) {
+            Node<T> node = new Node<>(head,value,tail);
+            for (int i = 1; i < index; i++) {
+                node.prev = node.prev.next;
+            }
+            node.prev.next = node;
+            for (int i = 1; i < size - index - 1; i++) {
+                node.next = node.next.prev;
+            }
+            node.next.prev = node;
+        }
+        if (index == 0 && size > 0) {
+            Node<T> node = new Node<>(null, value, head.next);
+            head.next.prev = node;
+            head = node;
+        }
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final T removedValue = get(index);
+        size--;
+        if (size == 0) {
+            head = null;
+        }
+        if (index == 0 && size > 0) {
+            head = head.next;
+            head.prev = null;
+        }
+        if (index == (size) && size > 0) {
+            tail = tail.prev;
+            tail.next = null;
+        }
+        if (index > 0 && index < (size)) {
+            Node<T> nodeBeforeIndex = head;
+            for (int i = 1; i < index; i++) {
+                nodeBeforeIndex = nodeBeforeIndex.next;
+            }
+            nodeBeforeIndex.next = nodeBeforeIndex.next.next;
+            Node<T> nodeAfterIndex = tail;
+            for (int i = 1; i < size - index; i++) {
+                nodeAfterIndex = nodeAfterIndex.prev;
+            }
+            nodeAfterIndex.prev = nodeAfterIndex.prev.prev;
+        }
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (node.item == object || (node.item != null && node.item.equals(object))) {
+                remove(i);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
 }
```

Comment:
According to common mistakes:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html

"Create a separate method unlink(Node node).
This logic will be used in 2 methods: when we remove from list by index and by value. So let’s create a method that will take a Node that needs to be removed and unlink it. As a result we will call it from both methods when Node is found."